src/Explorer/Explorer.tsx

```diff
@@ -393,9 +393,29 @@ export default class Explorer {
       useNotebook.getState().setConnectionInfo(connectionStatus);
       try {
         useNotebook.getState().setIsAllocating(true);
+        TelemetryProcessor.trace(Action.PhoenixConnection, ActionModifiers.Start, {
+          dataExplorerArea: Areas.Notebook,
+        });
         const connectionInfo = await this.phoenixClient.allocateContainer(provisionData);
-        await this.setNotebookInfo(connectionInfo, connectionStatus);
+        if (connectionInfo.status === HttpStatusCodes.OK && connectionInfo?.data?.notebookServerUrl) {
+          TelemetryProcessor.trace(Action.PhoenixConnection, ActionModifiers.Success, {
```

Comment:
To make sure we always do `traceStart` and `traceSuccess/traceFailure` in pair we should call `traceStart` as the first thing in try block, `traceSuccess` towards the end of try block and `traceFailure` in the catch block. That way we can guarantee we're not losing any traces.

Reply:
+1